loader/src/test/java/loader/TripleTableLoaderPartByPredTest.java

```diff
@@ -0,0 +1,462 @@
+package loader;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import org.apache.log4j.Logger;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Encoder;
+import org.apache.spark.sql.Encoders;
+import org.apache.spark.sql.Row;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import com.holdenkarau.spark.testing.JavaDataFrameSuiteBase;
+
+/**
+ * This class tests the parsing of the NT triples file and the building of the
+ * TripleTableLoader. It builds the triple table with a physical partitioning
+ * strategy. The table is partitioned by predicate.
+ * 
+ * @author Victor Anthony Arrascue Ayala
+ */
+public class TripleTableLoaderPartByPredTest extends JavaDataFrameSuiteBase implements Serializable {
```

Comment:
Generally, for me all the methods in this and the next class are connected with parsing the data, not with storing the data. So I do not see the point to repeat all the tests when partitioning by subject/predicate is involved. Moreover, you can always randomize the partitioning strategy and generalize all tests not to be only for default partitioning but for what it happens in the randomization. This way, you will cover the same cases when you have partitioning by subject/object with less repetition of tests.

Reply:
What do you mean by randomization?

Reply:
For example, a Boolean variable which randomly takes true values, randomly false values. Based on its value, you partition by subject.

Reply:
This is also not a good practice in JUnit tests.

Reply:
mmm... because randomly one can get all all tests passed and you can ruin someone's live :-) 
This is well known: https://en.wikipedia.org/wiki/Random_testing#Critique

Reply:
> because randomly one can get all all tests passed and you can ruin someone's live

Yes, u do not guarantee that you will cover all the cases at one run, but on the other hand, you limit the tests written (code) which is good in terms of maintaining the code. Anyway, my general comment was that parsing and partitioning are not really connected. Therefore, maybe we do not need the repetition of all those methods when partitioning is applied. They do not harm and in terms of maintainability at the moment, they are easy to maintain. But if more tests are added, this might be a problem. Yous decision at the end :)

Reply:
Yes. The problem is there is no traditional parsing process in the loader neither.
We read the data using a regular expression, remove faulty entries and save the data to another format. Therefore, the only way of checking whether the "parsing" was correct, is by checking the output. A traditional parser would either process the whole file if this is correct, or reject it, if there is one single error. This is not what we are doing, and for our goals, keeping as many well-formed triples as possible is what we need. Therefore, in the current project it is not possible to detach parsing from checking the outcome of the triple table, and this is reflected in the JUnit tests.